packages/dev/core/src/Engines/engine.ts

```diff
@@ -990,17 +990,17 @@ export class Engine extends ThinEngine {
         this._gl.deleteBuffer(buffer);
     }
 
-    private _clientWaitAsync(sync: WebGLSync, flags = 0, intervalms = 10): Promise<void> {
+    private _clientWaitAsync(sync: WebGLSync, flags = 0, checkForSyncInterval = 10): Promise<void> {
         const gl = <WebGL2RenderingContext>(this._gl as any);
         return new Promise((resolve, reject) => {
             const check = () => {
-                const res = gl.clientWaitSync(sync, flags, 0);
+                const res = gl.clientWaitSync(sync, flags, 0); // timeout = 0, checks SYNC state without blocking
                 if (res == gl.WAIT_FAILED) {
-                    reject();
+                    reject("WebGL sync object failed");

```

Comment:
reject(new Error("WebGL sync object failed"));

Reply:
Not really needed but let's have the stacktrace logged as well. 

Changed.